### PR TITLE
 Links Synchronizer to Video eliminating global scope

### PR DESCRIPTION
**Pull request by jaysaleh**
This pr links a `Synchronizer` object to a `Video` object eliminating global scope.

### CODE REVIEW

File: src/main/webapp/view/view.js [hunk 1]
```diff
@@ -43,12 +35,16 @@ getLecture().then((lecture) => {
...
 async function initialize() {
   setLectureName();
-  window.video.loadVideoApi();
-  intializeDiscussion();
+
+  const video = new Video();
+  const transcript = new TranscriptArea('event controller');
+
+  await video.loadVideoApi();
   // TODO: Move TranscriptArea initialization outside of initialize()
```
Comment (amuamushu) [inline src/main/webapp/view/view.js:43]: nit: maybe keep the TODO closer to where `TranscriptArea` was created.

File: src/main/webapp/view/video/video.js [hunk 2]
```diff
@@ -12,15 +12,22 @@
...
 // limitations under the License.
 
+import Synchronizer from '../../synchronizer.js';
 import {secondsToMilliseconds} from '../../timestamps.js';
 
 const SCRIPT = 'script';
 
 /** Initializes and stores video player information. */
 export default class Video {
+  /** Links a new `Synchronizer` to a `Video`. */
```
Comment (n-wach) [inline src/main/webapp/view/video/video.js:22]: Does this JSDoc need to be updated?

File: src/main/webapp/view/video/video.js [hunk 3]
```diff
@@ -12,15 +12,22 @@
...
+  constructor() {
+    this.synchronizer = new Synchronizer(this);
```
Comment (n-wach) [inline src/main/webapp/view/video/video.js:24]: can this.synchronizer be private?

File: src/main/webapp/view/video/video.js (hunk 2 above)
Comment (jaysaleh) [reply src/main/webapp/view/video/video.js:22]: rephrased

File: src/main/webapp/view/video/video.js (hunk 3 above)
Comment (jaysaleh) [reply src/main/webapp/view/video/video.js]: yes fixed

File: src/main/webapp/synchronizer.js [hunk 4]
```diff
@@ -29,13 +29,18 @@ const transcriptSeeker = new TranscriptSeeker('event controller');
  * time.
  */
 export default class Synchronizer {
+  /** Links a `Video` to a `Synchronizer`. */
```
Comment (billie-cat) [inline src/main/webapp/synchronizer.js:32]: nit: this constructor comment isn't necessary. I would omit.

File: src/main/webapp/view/video/video.js [hunk 5]
```diff
@@ -12,15 +12,24 @@
...
 
 const SCRIPT = 'script';
 
 /** Initializes and stores video player information. */
 export default class Video {
+  #synchronizer;
+
+  /** Creates a new `Synchronizer` linked to `this`. */
+  constructor() {
+    this.#synchronizer = new Synchronizer(this);
```
Comment (billie-cat) [inline src/main/webapp/view/video/video.js:26]: Typically it is bad design to have a circular dependency. Either the synchronizer should have a reference to Video or video have a reference to synchronizer. Can you refactor the code to remove this circular dependency.

Comment (jaysaleh) [reply src/main/webapp/view/video/video.js:25]: Modified to only have video have a reference to synchronizer

Comment (billie-cat) [reply src/main/webapp/view/video/video.js:25]: Why are you still passing 'this' to the Synchronizer. Can that be removed?

Comment (jaysaleh) [reply src/main/webapp/view/video/video.js:25]: Removed

File: src/main/webapp/view/video/video.js [hunk 6]
```diff
@@ -38,15 +46,16 @@ export default class Video {
       width: '640',
       videoId: window.LECTURE.videoId,
       events: {
-        onReady: this.onPlayerReady,
+        onReady: window.onPlayerReady,
```
Comment (billie-cat) [inline src/main/webapp/view/video/video.js:49]: which is this window now? the method is declared in this class.

Comment (jaysaleh) [reply src/main/webapp/view/video/video.js:49]: The window is needed for the API to be able to call this

Comment (billie-cat) [reply src/main/webapp/view/video/video.js:49]: talked offline and Jay explained why this is needed.

**Pull request merged**